assets/js/googlesitekit/datastore/user/authentication.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * core/user Data store: Authentication info.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+
+const { createRegistrySelector } = Data;
+
+// Actions
+const START_FETCH_AUTHENTICATION = 'START_FETCH_AUTHENTICATION';
+const FETCH_AUTHENTICATION = 'FETCH_AUTHENTICATION';
+const FINISH_FETCH_AUTHENTICATION = 'FINISH_FETCH_AUTHENTICATION';
+const CATCH_FETCH_AUTHENTICATION = 'CATCH_FETCH_AUTHENTICATION';
+const RECEIVE_AUTHENTICATION = 'RECEIVE_AUTHENTICATION';
+
+export const INITIAL_STATE = {
+	authentication: undefined,
+	isFetchingAuthentication: false,
+};
+
+export const actions = {
+	*fetchAuthentication() {
+		let response, error;
+
+		yield {
+			payload: {},
+			type: START_FETCH_AUTHENTICATION,
+		};
+
+		try {
+			response = yield {
+				payload: {},
+				type: FETCH_AUTHENTICATION,
+			};
+
+			yield actions.receiveAuthentication( response );
+
+			yield {
+				payload: {},
+				type: FINISH_FETCH_AUTHENTICATION,
+			};
+		} catch ( e ) {
+			error = e;
+			yield {
+				payload: { error },
+				type: CATCH_FETCH_AUTHENTICATION,
+			};
+		}
+
+		return { response, error };
+	},
+	/**
+	 * Stores connection info received from the REST API.
+	 *
+	 * @since 1.5.0
+	 * @private
+	 * @param {Object} authentication Authentication info from the API.
+	 * @return {Object} Redux-style action.
+	 */
+	receiveAuthentication( authentication ) {
+		invariant( authentication, 'authentication is required.' );
+
+		return {
+			payload: { authentication },
+			type: RECEIVE_AUTHENTICATION,
+		};
+	},
+};
+
+export const controls = {
+	[ FETCH_AUTHENTICATION ]: () => {
+		return API.get( 'core', 'user', 'authentication' );
```

Comment:
It looks like this wasn't part of the AC/IB but this route should be added to our preloaded paths (`googlesitekit_apifetch_preload_paths`).

We don't have any preloaded for core routes yet, so I'm not sure exactly where this should go yet, but perhaps `REST_Routes`. @felixarntz ?

Reply:
@aaemnnosttv I have not addressed yes but will once @felixarntz responds.

Reply:
Good point @aaemnnosttv (sorry, I had missed this ping earlier). The `core/site` `connection` route is currently not preloaded either though, so I think this is something we should revisit in a follow-up issue.